src/sentry/auth/helper.py

```diff
@@ -195,22 +195,25 @@ def handle_existing_identity(
         subdomain = None
         if data:
             subdomain = data.get("subdomain") or None
-        login_redirect_url = auth.get_login_redirect(self.request)
```

Comment:
The regression was that `auth.get_login_redirect(self.request)` here should not be called **_before_** `self._login(user)`.